DataManagementSystem/Agent/RequestOperations/ReplicateAndRegister.py

```diff
@@ -59,10 +57,16 @@ def filterReplicas( opFile, logger = None, dataManager = None, seCache = None ):
 
   replicas = replicas["Successful"].get( opFile.LFN, {} )
 
+  # FC checksum is more likely better than opFile.Checksum
+  fcMetadata = FileCatalog().getFileMetadata( opFile.LFN )
+  fcChecksum = fcMetadata.get( 'Value', {} ).get( 'Successful', {} ).get( opFile.LFN, {} ).get( 'Checksum', '' )
+  # Replace opFile.Checksum if it doesn't match a valid FC checksum
+  if fcChecksum and ( not opFile.Checksum or not compareAdler( fcChecksum, opFile.Checksum ) ):
+    opFile.Checksum = fcChecksum
```

Comment:
I can add a printout, but is it worth? I am pretty sure that if set, it is OK... The main use case for this test is when it is not set. Then we set it from the FC